File: 2021/src/test/java/de/jfriemel/aoc/days/Day09Test.java
package de.jfriemel.aoc.days;

import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.assertEquals;

public class Day09Test {

    private static final String INPUT_STRING = """
            2199943210
            3987894921
            9856789892
            8767896789
            9899965678""";

    private static final List<String> INPUT = Arrays.asList(INPUT_STRING.split("\n"));

    @Test
    public void testPart1() {
        String result = new Day09().part1(INPUT);
        assertEquals("15", result);
    }

    @Test
    public void testPart2() {
        String result = new Day09().part2(INPUT);
        assertEquals("1134", result);
    }

}

File: 2023/src/test/kotlin/de/jfriemel/aoc/days/Day09Test.kt
package de.jfriemel.aoc.days

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class Day09Test {
    private val input = """
        0 3 6 9 12 15
        1 3 6 10 15 21
        10 13 16 21 30 45
    """.trimIndent().lines()

    @Test
    fun testPart1() {
        val result = Day09.part1(input)
        Assertions.assertEquals("114", result)
    }

    @Test
    fun testPart2() {
        val result = Day09.part2(input)
        Assertions.assertEquals("2", result)
    }
}
